app/assets/javascripts/admin/bulk_order_management.js.coffee

```diff
@@ -162,6 +163,14 @@ angular.module("ofn.admin").controller "AdminOrderMgmtCtrl", [
       $scope.supplierFilter = $scope.suppliers[0].id
       $scope.orderCycleFilter = $scope.orderCycles[0].id
       $scope.quickSearch = ""
+
+    $scope.weightAdjustedPrice = (lineItem, oldValue) ->
+      if oldValue <= 0
+        oldValue = lineItem.units_variant.unit_value
+      if lineItem.unit_value <= 0
+        lineItem.unit_value = lineItem.units_variant.unit_value
+      lineItem.price = lineItem.price * lineItem.unit_value / oldValue
+      #$scope.bulk_order_form.line_item.price.$setViewValue($scope.bulk_order_form.line_item.price.$viewValue)
 ]
 
```

Comment:
Ok, don't we need to make sure that oldValue is not zero? I think that is what is causing the issue in my system.

With respect to migrating the data, I suppose we need to have a discussion about how exactly the unit_value field on line_item will be used. If we are going to need it in other pages (say reports), and we are NOT going to require that every line item has a unit_value, then we are going to have to write some logic to work out which unit_value is reported (that of the line_item itself, or that of its variant).

I think that based on my (probably limited) understanding of what this field is doing, my strong preference would be for us to have confidence that we can always read the unit_value from the line_item, and that will give us all of the information we need to know without ever having to look at the original variant. That way our data would behave consistently and in a predictable manner. I also like it because it more of less copies the pattern of the price field on line_item, which is completely abstracted from the price of the original variant and is used as the source of truth about the price the line_item was sold for.

I am happy to write the migration - I think it will just be a matter of cycling through all of the line items in the system and copying the unit value across. If you have a strong feeling another way, let me know. :)